### PR TITLE
Implement database based on diagram

### PR DESCRIPTION
**Pull request by GhausAliKhan**
### Description
This pull request addresses the project requirements of creating a database based on a given clinic database diagram. The primary goal is to interpret the diagram and implement the necessary SQL code to construct the database accurately.

**Note: Below are the screenshots attached for the reason I'm working solo.**
image
image
image

**Tasks:**
- Created the file schema_based_on_diagram.sql to implement the database structure as per the diagram.
- Added queries to handle join tables for many-to-many relationships that may not appear explicitly in the diagram.
- Ensured the addition of foreign key indexes to relevant tables for data integrity.
- Included the clinic database diagram image in the project's README for reference.
- The database schema is accurately implemented, and all necessary components are in place to support the clinic database.

Your review and feedback are highly appreciated.

### CODE REVIEW

File: schema_based_on_diagram.sql [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+
+-- INVOICES - MEDICAL_HISTORIES - PATIENTS
+SELECT patients.name as PatientName, medical_histories.admitted_at as Admitted, invoices.payed_at as Payed FROM invoices
+JOIN medical_histories ON invoices.medical_history_id = medical_histories.id
+JOIN patients ON medical_histories.patient_id = patients.id;
+
+/* TREATMENTS - INVOICES - INVOICE_ITEMS */
+SELECT treatments.name as treatment, invoice_items.quantity, invoices.total_amount FROM invoice_items
+JOIN treatments ON invoice_items.treatment_id = treatments.id
+JOIN invoices ON invoice_items.invoice_id = invoices.id;
```
Comment (1GORDON) [inline schema_based_on_diagram.sql]: Great job @GhausAliKhan 🤝

- [x] As stated in the project requirements, kindly ensure all "join table" for the many-to-many relationships is created, even if they are not shown on the diagram. You may kindly create a table between `medical_histories` and `treatments` to achieve this.

medical-histories

File: schema_based_on_diagram.sql [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+    name VARCHAR(100)?
+);
+
+CREATE TABLE invoices(
+    id SERIAL PRIMARY KEY,
+    total_amount DECIMAL,
+    generated_at TIMESTAMP,
+    payed_at TIMESTAMP,
+    medical_histories_id INT
+);
```
Comment (1GORDON) [inline schema_based_on_diagram.sql]: Great job @GhausAliKhan 🤝

- [x] As stated in the project requirements, kindly ensure all FOREIGN KEYS are set. kindly add all the missing FOREIGN KEYS.
foreign-keys

File: schema_based_on_diagram.sql [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+    id SERIAL PRIMARY KEY,
+    unit_price DECIMAL,
+    quantity INT,
+    total_price DECIMAL,
+    invoice_id INT,
+    treatment_id INT,
+    FOREIGN KEY(invoice_id) REFERENCES invoices(id),
+    FOREIGN KEY(treatment_id) REFERENCES treatments(id)
+);
+
```
Comment (1GORDON) [inline schema_based_on_diagram.sql:44]: Great job @GhausAliKhan 🤝

- [x] As stated in the project requirements, kindly ensure all FOREIGN KEYS have indexes created.
indexes

**Review by 1GORDON — CHANGES_REQUESTED** [review]
##
## Hi @GhausAliKhan ✋

Good job so far!

## Highlights!
- All tables from the diagram are created (by using the `CREATE TABLE` command) ✔️ 
- All tables have a PRIMARY KEY set ✔️

Hello ✋, you have done well reaching this point of the milestone 🤝 . But there are some issues that you still need to work on to go to the next project but you are almost there!

# Status: Required Changes ♻️⛔️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend considering them as they can improve your code._

Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread and tag me `@1GORDON` if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think the code review was unfair, request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by ghost — APPROVED** [review]
Hi @GhausAliKhan , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

YesYouAreGIF (2)



## Highlights
✅ All foreign keys are set
✅ All foreign keys have indexes 👍 

## Optional suggestions

N/A

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**